File: kotlin/src/main/kotlin/signer.kt
package io.github.wulkanowy.signer

import java.io.ByteArrayInputStream
import java.security.KeyStore
import java.security.PrivateKey
import java.security.Signature
import java.util.*

fun signContent(password: String, certificate: String, content: String): String? {
    val keystore = KeyStore.getInstance("pkcs12").apply {
        load(ByteArrayInputStream(Base64.getDecoder().decode(certificate)), password.toCharArray())
    }
    val signature = Signature.getInstance("SHA1WithRSA").apply {
        initSign(keystore.getKey("LoginCert", password.toCharArray()) as PrivateKey)
        update(content.toByteArray())
    }
    return Base64.getEncoder().encodeToString(signature.sign())
}

File: php/src/signer.php
<?php

namespace Wulkanowy\UonetRequestSigner;

function signContent(string $password, string $certificate, string $content)
{
    openssl_pkcs12_read(base64_decode($certificate), $keystore, $password);

    if (null === $keystore) {
        throw new \RuntimeException('Invalid password');
    }

    $privateKey = openssl_pkey_get_private($keystore['pkey'], $password);

    openssl_sign($content, $signature, $privateKey, OPENSSL_ALGO_SHA1);

    return base64_encode($signature);
}
